README.md

```diff
@@ -8,9 +8,82 @@ The Harvester Container Storage Interface (CSI) Driver provides a [CSI](https://
 `make`
 
 
-## Running
+## Deploying
 
-`./bin/harvester-csi-driver`
+### Prerequisites
+
+- The Kubernetes cluster is built on top of Harvester virtual machines.
+- The Harvester virtual machines running as guest Kubernetes nodes are in the same namespace.
+
+### Deploying with Harvester node driver
+
+When spin up a kubernetes cluster using Rancher RKE2 node driver, the Harvester CSI driver will be deployed when harvester cloud provider is selected.
+
+### Deploying manually
+
+1. Prepare a kubeconfig file of the host Harvester cluster
+
+Before running the following scripts, kubectl should point to the host Harvester cluster.
+```
+NAMESPACE=namespace-of-vms
+SERVICE_ACCOUNT_NAME=example-sa
+ROLE_BINDING_NAME=example-rolebinding
+TARGET_DIR=./example
+mkdir -p ${TARGET_DIR}
+# Create a service account
+kubectl create sa ${SERVICE_ACCOUNT_NAME} --namespace ${NAMESPACE}
+
+# Create a rolebinding
+kubectl create rolebinding ${ROLE_BINDING_NAME} --serviceaccount=${NAMESPACE}:${SERVICE_ACCOUNT_NAME} --clusterrole=harvesterhci.io:cloudprovider
+
+# Get kube config fields
+SECRET_NAME=$(kubectl get sa "${SERVICE_ACCOUNT_NAME}" --namespace="${NAMESPACE}" -o jsonpath="{.secrets[].name}")
+kubectl get secret --namespace "${NAMESPACE}" "${SECRET_NAME}" -o jsonpath="{.data.ca\.crt}" | base64 -d >"${TARGET_DIR}/ca.crt"
+SA_TOKEN=$(kubectl get secret --namespace "${NAMESPACE}" "${SECRET_NAME}" -o jsonpath="{.data.token}" | base64 -d)
+CURRENT_CONTEXT=$(kubectl config current-context)
+CLUSTER_NAME=$(kubectl config get-contexts "${CURRENT_CONTEXT}" | awk '{print $3}' | tail -n 1)
+ENDPOINT=$(kubectl config view -o jsonpath="{.clusters[?(@.name == \"${CLUSTER_NAME}\")].cluster.server}")
+
+# create kubeconfig file
+GENERATED_KUBECONFIG="${TARGET_DIR}/kube.config"
+kubectl config set-cluster "${CLUSTER_NAME}" \
+    --kubeconfig="${GENERATED_KUBECONFIG}" \
+    --server="${ENDPOINT}" \
+    --certificate-authority="${TARGET_DIR}/ca.crt" \
+    --embed-certs=true
+kubectl config set-credentials \
+    "${SERVICE_ACCOUNT_NAME}-${NAMESPACE}-${CLUSTER_NAME}" \
+    --kubeconfig="${GENERATED_KUBECONFIG}" \
+    --token="${SA_TOKEN}"
+kubectl config set-context \
+    "${SERVICE_ACCOUNT_NAME}-${NAMESPACE}-${CLUSTER_NAME}" \
+    --kubeconfig="$${GENERATED_KUBECONFIG}" \
+    --cluster="${CLUSTER_NAME}" \
+    --user="${SERVICE_ACCOUNT_NAME}-${NAMESPACE}-${CLUSTER_NAME}" \
+    --namespace="${NAMESPACE}"
+kubectl config use-context "${SERVICE_ACCOUNT_NAME}-${NAMESPACE}-${CLUSTER_NAME}" \
+    --kubeconfig="${GENERATED_KUBECONFIG}"
+```
```

Comment:
can we reuse this script here so it is much user-friendly https://github.com/harvester/cloud-provider-harvester/blob/master/deploy/generate_addon.sh, thanks

Reply:
It's not general to be reused directly. I add an updated one here.